egs/wsj/s5/utils/validate_lang.pl

```diff
@@ -814,22 +829,42 @@ sub check_summation {
       next;
     }
     $wlen = int(rand(100)) + 1;
+    $end_subword = 0;
     print "--> generating a $wlen word sequence\n";
     $wordseq = "";
     $sid = 0;
     $wordseq_syms = "";
-    foreach (1 .. $wlen) {
+    $count = 0;
+    while ($count < $wlen) {
       $id = int(rand(scalar(keys %wint2sym)));
       # exclude disambiguation symbols, BOS and EOS, epsilon, and
       # grammar-related symbols from the word sequence.
-      while (defined $wdisambig_words_hash{$id} or
+      $subword = $wint2sym{$id};
+      $suffix = substr($subword, -2, 2);
```

Comment:
are you assuming the separator has 2 characters here?  Seems unsafe.  and I hope this doesn't crash for words shorter than 2 characters.

Reply:
Yes, you are right. I will fix that.